static_report/src/utils/index.tsx

```diff
@@ -134,3 +136,113 @@ export function extractExpectedOrActual(value) {
 
   return value;
 }
+
+// for comparison
+export function joinBykey(base = {}, input = {}) {
+  const result = {};
+
+  Object.entries(base).forEach(([key, value]) => {
+    if (!result[key]) {
+      result[key] = {};
+    }
+
+    result[key]['base'] = value;
+  });
+
+  Object.entries(input).forEach(([key, value]) => {
+    if (!result[key]) {
+      result[key] = {};
+    }
+    result[key]['input'] = value;
+  });
+
+  return result;
+}
+
+export function getComparisonTests(assertion, from) {
+  const { passed, failed } = getReportAsserationStatusCounts(assertion);
+
+  if (assertion) {
```

Comment:
I think checking for `!assertion` (base case) up top first would make this more readable